File: Leetcode/python/711-jewels-and-stones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        res = 0
        jewelMap = {}

        for c in jewels:
            jewelMap[c] = jewelMap.get(c, 0)

        for c in stones:
            if c in jewelMap:
                jewelMap[c] += 1

        for key in jewelMap.keys():
            res += jewelMap[key]

        return res

File: Leetcode/typescript/711-jewels-and-stones.ts
function numJewelsInStones(jewels: string, stones: string): number {
  let jewelMap = new Set([...jewels]);
  let res = 0;
  for (const c of stones) {
    if (jewelMap.has(c)) {
      res += 1;
    }
  }
  return res;
}
